app/src/service/ocr/index.ts

```diff
@@ -1,35 +1,138 @@
+import { createLogger } from "@/service/factory";
+import { type ParserKeys, parseOcrResult, parsers } from "@/service/ocr/parser";
+import {
+  DocumentImage,
+  DocumentOrientation,
+  orientations,
+  rotateDocumentImage,
+} from "@/utils/document";
 import { createWorker } from "tesseract.js";
 
-import { createLogger } from "../../utils/logger";
-import * as parsers from "./parser";
+export type DocumentMatcher<
+  K extends ParserKeys,
+  P extends Record<string, RegExp> = Record<string, RegExp>
+> = {
+  name: string;
+  id: K;
+  patterns: P;
+};
 
-type OcrOptions = {
+export type OcrOptions = {
   debug: boolean;
 };
 
+export type ProcessedImageResult = {
+  documents: ReturnType<typeof parseOcrResult>;
+  image: DocumentImage;
+  percentages: Record<ParserKeys, number>;
+  confidence: number;
+  rotatedOrientation?: DocumentOrientation;
+};
+
+export type ProcessedRotatedImagesResult = ProcessedImageResult[];
+
+export type TextFromImagePathResult = {
+  text: string;
+  confidence: number;
+};
+
 // main logger
 export const logger = createLogger("ocr-parser", {
-  enabled: process.env.LOG_LEVEL === "debug",
+  enabled: false,
   mixin: () => {
     return { module: "ocr-parser" };
   },
 });
 
-const scanImage = async (path: string, opts: OcrOptions) => {
+const getTextFromImagePath = async (
```

Comment:
I've made a note to change the name of this function to `getTextFromDocument` since it actually runs the OCR against the `data` property of the `DocumentImage` since the `data` property holds a reference to rotated document images. 

Otherwise we'd have to write a tmp rotated file to desk and pass its path. I've found that working with files that are written/modified/deleted during a short-lived process could lead to file lock issues.